src/authenticators/jwt_svid_authenticator/mod.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+//! JWT SVID authenticator
+
+use super::ApplicationName;
+use super::Authenticate;
+use crate::front::listener::ConnectionMetadata;
+use log::error;
+use parsec_interface::operations::list_authenticators;
+use parsec_interface::requests::request::RequestAuth;
+use parsec_interface::requests::Result;
+use parsec_interface::requests::{AuthType, ResponseStatus};
+use parsec_interface::secrecy::ExposeSecret;
+use spiffe::svid::jwt::Jwt;
+use spiffe::workload::jwt::JWTClient;
+use std::str;
+use std::time::Duration;
+
+/// JWT SVID authenticator
+#[allow(missing_debug_implementations)]
+pub struct JwtSvidAuthenticator {
+    jwt_client: JWTClient,
+}
+
+impl JwtSvidAuthenticator {
+    /// Create a new JWT-SVID authenticator with a specific path to the Workload API socket.
+    pub fn new(workload_endpoint: String) -> Self {
+        JwtSvidAuthenticator {
+            jwt_client: JWTClient::new(&workload_endpoint, None),
+        }
+    }
+}
+
+impl Authenticate for JwtSvidAuthenticator {
+    fn describe(&self) -> Result<list_authenticators::AuthenticatorInfo> {
+        Ok(list_authenticators::AuthenticatorInfo {
+            description: String::from(
+                "Authenticator validating a JWT SPIFFE Verifiable Identity Document",
+            ),
+            version_maj: 0,
+            version_min: 1,
+            version_rev: 0,
+            id: AuthType::JwtSvid,
+        })
+    }
+
+    fn authenticate(
+        &self,
+        auth: &RequestAuth,
+        _: Option<ConnectionMetadata>,
+    ) -> Result<ApplicationName> {
+        let svid = Jwt::new(
+            str::from_utf8(auth.buffer.expose_secret())
+                .map_err(|e| {
+                    error!(
+                        "The authentication buffer can not be parsed into a UTF-8 string ({}).",
+                        e
+                    );
+                    ResponseStatus::InvalidEncoding
+                })?
+                .to_string(),
```

Comment:
(Asking because for some reason I thought this might be a JWT base64 encoded)

Reply:
Ah, I read about it now from the book PR. I think it's safe to assume `Jwt::new` would return a `Result` since it has to verify it's in the correct encoding? Tbh, maybea method like `Jwt::from_compact_serialization` would be more correct

Reply:
Yes agree, ideally the `Jwt` constructor would parse the Base 64-url encoded token and verify that the format is good. I used a very simple approach here where the Rust SPIFFE library does not do anything fancy. If the token is malformed, I expect the validation step to fail and return an error.